Change Log Severity from Debug for pipeline or step creation errors

flimmy opened the pull request:
Hey Everyone,

i would like to propose a change to the log level of several "debug" messages, as the debug log level is not suitable for a production setup and some IMHO important information gets lost.
As the perceived importance is inherently subjective any input is welcome.
Here are my 2 cents:

| message                                                            | prev. Lvl | new Lvl | Reasoning                                                                  |
|--------------------------------------------------------------------|-----------|---------|----------------------------------------------------------------------------|
| ignoring pipeline as skip-ci was found ...                         | debug     | info    | skips are not visible in the browser, so make them visible in log          |
| cannot find config '%s' in '%s' with ...                           | debug     | info    | no config maybe a misconfigured lookup, so make it visible to the sysadmin |
| ~~failed to parse yaml~~                                           | debug     | warn    | ~~a failure should be at least a warning~~                                 |
| ErrFiltered (ignoring hook: 'when' filters filtered out all steps) | debug     | info    | the most common request i get as a sysadmin "why does it not start"        |
| marked as skipped, does not match metadata                         | debug     | info    | same as above, but more context                                            |
| ~~pipeline config could not be parsed~~                            | debug     | warn    | ~~a failure should be at least a warning~~                                 |

Edit 1: strike the 2 proposed warning changes as these are already shown in the web-ui

lafriks commented:
I also agree that most of these should be UI errors

6543 commented:
I also agree that instead of spaming the logs we should have it propergated to the UI in a nice way

flimmy commented:
rebased to upstream and reverted the loglevel change for the two errors that already generate a message in the webui.

Further research on the remaining 4 changes:
| message                                                            | line                                                                                                                                       | in-code reason for silent drop                                             |
|--------------------------------------------------------------------|--------------------------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------------------------------|
| ignoring pipeline as skip-ci was found ...                         | [51](https://github.com/flimmy/woodpecker/blob/0cf0098b0dd2731e1ce73e6279b5a46d4f16f791/server/pipeline/create.go#L51)                     | `return nil, ErrFiltered` instead of a pipeline with error like in line 97 |
| cannot find config '%s' in '%s' with ...                           | [83](https://github.com/flimmy/woodpecker/blob/0cf0098b0dd2731e1ce73e6279b5a46d4f16f791/server/pipeline/create.go#L83)                     | `return nil, ErrFiltered` instead of a pipeline with error like in line 97 |
| ErrFiltered (ignoring hook: 'when' filters filtered out all steps) | [103](https://github.com/flimmy/woodpecker/blob/0cf0098b0dd2731e1ce73e6279b5a46d4f16f791/server/pipeline/create.go#L103)                   | `return nil, ErrFiltered` instead of a pipeline with error like in line 97 |
| marked as skipped, does not match metadata                         | [165](https://github.com/flimmy/woodpecker/blob/0cf0098b0dd2731e1ce73e6279b5a46d4f16f791/server/pipeline/stepbuilder/step_builder.go#L165) | `return nil, nil` instead of appending an error                            |

If returning the pipeline and an error for the messages in server/pipeline/create.go or generating and returning an err in server/pipeline/stepbuilder/step_builder.go is a "quick fix" for one of you, or something you would see yourself working on in the near future, please let me know, else i will try to find some time to have a deeper look into the codebase to undestand the implications of creating incomplete pipeline objects.

Or maybe we can merge the current loglevel changes and i create a seperate issue to "add ui warnings for silently dropped pipelines".

flimmy commented:
> I would be against this as this would result in a large amount of unneeded logs for large instances, if needed this should be added maybe as an option that by default is off

what exactly?: the increased log level, or the addition of a ui warning and thus a database entry

If its the first: i would like to argue that log-level "info" is not suitable for large installations anyway (in my experience error or warn are being used), and since there is nothing between info and debug (which is not suitable for any production setup no matter the size), i personally feel like "info" is a good compromise to silently dropping it and thus generating work for people trying to figure out why their pipeline wont start.
